### PR TITLE
Add file with build revision

### PR DESCRIPTION
**Pull request by holzingk**

### CODE REVIEW

**Comment by c-mauderer** [comment]
I don't really have a problem with merging this but just out of curiosity: What is the purpose of that file?

If for example you want to use it in some c file for printing a revision, it would be a better solution to directly generate a header (maybe in libgrisp).

**Comment by holzingk** [comment]
I tend to forget which revision I build my toolchain with.

We can also read it with our Erlang build tools, to warn users if something changed in RTEMS so that they rebuild Erlang maybe.

I don't know if keeping track of the build revision is necessary at all, or if it can be obtained otherwise.

Writing it in a C header could be valuable as well, but it's hard to access that value from Erlang during buildtime.

Maybe keep that open and wait for other people's comments?

**Comment by c-mauderer** [comment]
If it's just a reminder for the toolchain version, the file should be generated in the rtems-install directory.

**Comment by eproxus** [comment]
+1 to this change. Both a file and header seems useful. I think reading it from a header file shouldn't be a big problem, ever from Erlang

**Comment by c-mauderer** [comment]
OK. So basically the following changes would be good before merging that patch:
- Create the file in `rtems-install`
- (optional because there is currently no use case) generate a header file like `rtems-install/rtems/4.12/arm-rtems-4.12/atsamv/lib/include/grisp/build-version.h`.

**Comment by holzingk** [comment]
Okay. Will update PR.

I could also create a `.hrl` file.

**Comment by eproxus** [comment]
@nextl00p Create a `.hrl` file as well, maybe we can find a use for it

File: build/build.sh [hunk 1]
```diff
@@ -62,3 +62,5 @@ cd "${PROJECTDIR}"
 "${SCRIPTDIR}/build-libbsd.sh" clean install
 "${SCRIPTDIR}/build-libinih.sh"
 "${SCRIPTDIR}/build-libgrisp.sh"
+
+git log -1 --format="%H" > buildrev
```
Comment (eproxus) [inline build/build.sh]: We should use `git rev-parse HEAD` here instead of `git log`. `git log` isn't made for programmatic access like this.

**Comment by holzingk** [comment]
Updated!

**Comment by c-mauderer** [comment]
Looks mostly OK for me. Two points:
- I think the entry in `.gitignore` is no longer necessary with the new version of the patch.
- I'm really not sure whether the headers in `${PREFIX}/include` can be used by an application. Try whether you can include it in some of the applications and still compile it. `${PREFIX}/${TARGET}/${BSP_NAME}/lib/include/grisp` would be an alternative path for that.

**Pull request merged**